src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -0,0 +1,227 @@
+            };
+
+            if (!_memoryActivities.TryAdd(memoryActivity))
+            {
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher: Buffer full. Dropping current batch of memory activities"));
+                DrainActivities(_currentMemoryActivities, _memoryActivities);
+            }
+        }
+
+        private void DrainActivities<T>(ConcurrentQueue<T> currentActivites, BlockingCollection<T> activitiesToProcess)
+        {
+            currentActivites.Clear();
+
+            if (activitiesToProcess.Any())
+            {
+                while (activitiesToProcess.TryTake(out T item))
+                {
+                    currentActivites.Enqueue(item);
+                }
+            }
+        }
+
+        internal void OnFunctionMetricsPublishTimer(object state)
+        {
+            PublishActivity(_currentMemoryActivities, _memoryActivities, PublishMemoryActivityPath);
+            PublishActivity(_currentFunctionActivities, _functionActivities, PublishFunctionActivityPath);
+        }
+
+        private void PublishActivity<T>(ConcurrentQueue<T> currentActivities, BlockingCollection<T> activitiesToProcess, string publishPath)
+        {
+            try
+            {
+                if (currentActivities.Any())
+                {
+                    SendRequest(currentActivities, publishPath);
+                }
+                DrainActivities(currentActivities, activitiesToProcess);
+            }
+            catch (Exception ex)
+            {
+                _errorCount++;
+                _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher : Error publishing activites. Exception : {0}", ex.ToString()));
+                if (_errorCount < _maxErrorLimit)
+                {
+                    _logTraceEvent(LogLevel.Warning, string.Format("LinuxContainerMetricsPublisher : Error count {0} within threshold. will retry later.", _errorCount));
+                }
+                else
+                {
+                    _logTraceEvent(LogLevel.Error, string.Format("LinuxContainerMetricsPublisher : Max error limit reached. Draining activities for {0}", _containerName));
+                    DrainActivities(currentActivities, activitiesToProcess);
+                    _errorCount = 0;
+                }
+            }
+        }
+
+        private void SendRequest<T>(ConcurrentQueue<T> activitiesToPublish, string publishPath)
+        {
+            var request = BuildRequest(HttpMethod.Post, publishPath, activitiesToPublish.ToArray());
+            HttpResponseMessage response = _httpClient.SendAsync(request).GetAwaiter().GetResult();
+            response.EnsureSuccessStatusCode();
+        }
+
+        public void InitializeRequestHeaders()
+        {
+            _hostName = _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteHostName);
```

Comment:
Use HostNameProvider for this - it handles slot swaps correctly. See other usages.